cozy/view_model/search_view_model.py

```diff
@@ -1,75 +1,74 @@
-import cozy.ext.inject as inject
+from typing import Callable
 
-from cozy.extensions.set import split_strings_to_set
-from cozy.open_view import OpenView
+from gi.repository import GLib
+
+import cozy.ext.inject as inject
 from cozy.application_settings import ApplicationSettings
 from cozy.architecture.event_sender import EventSender
 from cozy.architecture.observable import Observable
 from cozy.control.filesystem_monitor import FilesystemMonitor
+from cozy.extensions.set import split_strings_to_set
 from cozy.model.book import Book
 from cozy.model.library import Library
+from cozy.open_view import OpenView
 
 
 class SearchViewModel(Observable, EventSender):
     _fs_monitor: FilesystemMonitor = inject.attr("FilesystemMonitor")
     _model: Library = inject.attr(Library)
     _application_settings: ApplicationSettings = inject.attr(ApplicationSettings)
 
-    _search_open: bool = False
-
     def __init__(self):
         super().__init__()
         super(Observable, self).__init__()
 
-    @property
-    def books(self):
-        return self._model.books
-
-    @property
-    def authors(self):
+    def _get_available_books(self) -> set[Book]:
```

Comment:
Do we need `_get_available_books` to return a set? I.e. can `self._model.books` contain duplicates or some other property of a set is used? If not and we could use `list[Book]`, we could simply return `self._model.books` if `show_offline_books` is True (avoid building a set) or do the list comprehension with filtering otherwise.

Reply:
Good point. We don't. I just copied the code from the authors getter, and saw no reason to use a list, if we're building a new iterable anyway, but it's indeed better to check `show_offline_books` beforehand, and don't build anything if it's True,

Reply:
Done